packages/app/src/cli/services/execute-operation.ts

```diff
@@ -11,34 +11,66 @@ import {AbortError} from '@shopify/cli-kit/node/error'
 import {adminRequestDoc} from '@shopify/cli-kit/node/api/admin'
 import {ClientError} from 'graphql-request'
 import {parse} from 'graphql'
-import {writeFile} from '@shopify/cli-kit/node/fs'
+import {writeFile, readFile, fileExists} from '@shopify/cli-kit/node/fs'
 
 interface ExecuteOperationInput {
   organization: Organization
   remoteApp: OrganizationApp
   storeFqdn: string
   query: string
   variables?: string
+  variableFile?: string
   outputFile?: string
   version?: string
 }
 
-async function parseVariables(variables?: string): Promise<{[key: string]: unknown} | undefined> {
-  if (!variables) return undefined
-
-  try {
-    return JSON.parse(variables)
-  } catch (error) {
-    const errorMessage = error instanceof Error ? error.message : 'Unknown error'
-    throw new AbortError(
-      outputContent`Invalid JSON in ${outputToken.yellow('--variables')} flag: ${errorMessage}`,
-      'Please provide valid JSON format.',
-    )
+async function parseVariables(
+  variables?: string,
+  variableFile?: string,
+): Promise<{[key: string]: unknown} | undefined> {
+  if (variables) {
```

Comment:
similar comment here, I think we should error when someone tries to pass in both `variables` and `variableFile`